### PR TITLE
Fix HTTP config errors #268

### PR DESCRIPTION
**Pull request by maia-iyer**
Current image finishes immediately and does not properly wait for go subroutines. 
Also fixes HCL parse error and adds documentation

### CODE REVIEW

File: docs/conf/agent/full.conf [hunk 1]
```diff
@@ -8,27 +8,17 @@ server {
...
-    enabled = true
-    port = 30000                  # container port for mTLS connection
-    cert = "sample-keys/tls.pem"  # mTLS cert
-    key = "sample-keys/key.pem"   # mTLS key
-    ca = "sample-keys/rootCA.pem" # mTLS CA
+  # [optional, recommended] configure HTTPS connection to Tornjak server
+  https {
+    port = 10443                  # [required] container port for HTTPS connection
+    cert = "sample-keys/tls.pem"  # [required] TLS cert
+    key = "sample-keys/key.pem"   # [required] TLS key
```
Comment (mrsabath) [inline docs/conf/agent/full.conf:20]:
```suggestion
    key = "sample-keys/key.pem"   # [required for HTTPS] TLS key
```

File: docs/config-tornjak-server.md [hunk 2]
```diff
@@ -41,25 +41,17 @@ server {
 
     spire_socket_path = "unix:///tmp/spire-server/private/api.sock" # socket to communicate with SPIRE server
 
-    http {
-        enabled = true # if true, opens HTTP. if false, no HTTP connection opened
-	    port = "10000" # if HTTP enabled, opens HTTP listen port at container port 10000
+    http { # required block
+	    port = 10080 # if HTTP enabled, opens HTTP listen port at container port 10000
```
Comment (mrsabath) [inline docs/config-tornjak-server.md:45]:
```suggestion
	    port = 10080 # if HTTP enabled, opens HTTP listen port at container port 10080
```

File: docs/config-tornjak-server.md [hunk 3]
```diff
@@ -41,25 +41,17 @@ server {
...
     }
 
-    tls {
-        enabled = true # if true, opens TLS. if false, no TLS connection opened
-        port = "20000" # if enabled, opens TLS listen port at container port 20000
+    https {
+        port = 10443 # if enabled, opens TLS listen port at container port 20000
```
Comment (mrsabath) [inline docs/config-tornjak-server.md:49]:
```suggestion
        port = 10443 # if enabled, opens HTTPS listen port at container port 10443
```

File: docs/config-tornjak-server.md [hunk 4]
```diff
@@ -41,25 +41,17 @@ server {
...
         cert = "sample-keys/tls.pem" # path of certificate for TLS
         key = "sample-keys/key.pem" # path of keys for TLS
+        ca = "sample-keys/rootCA.pem"
```
Comment (mrsabath) [inline docs/config-tornjak-server.md:52]:
```suggestion
        ca = "sample-keys/userCA.pem" # [optional, enables mTLS] User CA 
```

File: docs/quickstart/tornjak-configmap.yaml [hunk 5]
```diff
@@ -13,9 +13,9 @@ data:
 
       # configure HTTP connection to Tornjak server
       http {
-        enabled = true
-        port = 10000 # opens at port 10000
+        port = 10080 # opens at port 10000
```
Comment (mrsabath) [inline docs/quickstart/tornjak-configmap.yaml:16]:
```suggestion
        port = 10080 # opens at port 10080
```

File: tornjak-backend/api/agent/server.go [hunk 6]
```diff
@@ -633,8 +633,8 @@ func (s *Server) HandleRequests() {
 	}()
 
 	if serverConfig.HTTPSConfig != nil {
+		numPorts += 1
 		go func() {
-			numPorts += 1
 			if serverConfig.HTTPSConfig.ListenPort == 0 {
 				serverConfig.HTTPSConfig.ListenPort = 443
```
[the remark this answers is not in the document]
Comment (mrsabath) [reply tornjak-backend/api/agent/server.go:639]: ok, let's keep the default 443 then

[the remark this answers is not in the document]
Comment (maia-iyer) [reply tornjak-backend/api/agent/server.go:639]: I think i will change it to error - but i'll do this with the next PR having to to do with HTTPS port

**Review by mrsabath — APPROVED** [review]
Thank you @maia-iyer for a fast fix! LGTM

**Pull request merged**